src/player/Marine.cpp

```diff
@@ -72,3 +72,120 @@ int32_t Marine::checkForPickUp() {
     }
     return -1;
 }
+
+/**
+* Date: Mar 27 
+* Author: Aing Ragunathan
+*
+* Interface: void Marine::updateImageDirection()
+*
+* Description:
+*       This function changes the direction that the character is facing.
+*       It is called from GameStateMatch::handle after every frame and
+*       updates the direction of the sprite according to the angle of the
+*       mouse from the center of the screen.
+*/
+void Marine::updateImageDirection() {
+    double angle = getAngle();
+    double radians = (angle -90) * M_PI/180;
+    
+    //order: start from ~0 rad, counter clockwise
+    if (radians > -M_PI/8 && radians < M_PI/8) { 
+        Entity::setSpritePosY(SPRITE_SIZE_Y*2); //right
+    }
+    else if (radians > -3*M_PI/8 && radians < -M_PI/8) { 
+        Entity::setSpritePosY(SPRITE_SIZE_Y*3);  //back & right
+    }
+    else if (radians > -5*M_PI/8 && radians < -3*M_PI/8) { 
+        Entity::setSpritePosY(SPRITE_SIZE_Y*4);  //back
+    }
+    else if (radians > -7*M_PI/8 && radians < -5*M_PI/8) { 
+        Entity::setSpritePosY(SPRITE_SIZE_Y*5); //back & left
+    }
+    else if (radians > -9*M_PI/8 && radians < -7*M_PI/8) { 
+        Entity::setSpritePosY(SPRITE_SIZE_Y*6); //left
+    }
+    else if (radians > -11*M_PI/8 && radians < -9*M_PI/8) { 
+        Entity::setSpritePosY(SPRITE_SIZE_Y*7); //front & left
+    }
+    else if (radians < -11*M_PI/8) { 
+        Entity::setSpritePosY(0); //front 
+    }
+    else if (radians > M_PI/8 && radians < 3*M_PI/8) { 
+        Entity::setSpritePosY(SPRITE_SIZE_Y); //front & right
+    }
+
+    setSrcRect(Entity::getSpritePosX(), Entity::getSpritePosY(), 75, 125); //update image
+}
+
+/**
+* Date: Mar 30 
+* Author: Aing Ragunathan
+*
+* Function Interface: void Marine::updateImageWalk(double frameCount)
+*       double frameCount - counted frames while walking
+*
+* Description:
+*       This function repeatedly updates the image of the marine in order to animate 
+*       walking. It is called from GameStateMatch::handle after every frame and updates
+*       the feet placement of the marine.
+*/
+void Marine::updateImageWalk(const Uint8 *state, double frameCount) {
+    if (state[SDL_SCANCODE_UP] || state[SDL_SCANCODE_W]) {
+        if (getSpritePosX() == 0) {
+            setSpritePosX(SPRITE_SIZE_X);   //stops lag when taking first step
+        }
+        else if ((int)frameCount % FRAME_COUNT_WALK == 0) {
+            if (getSpritePosX() < SPRITE_SIZE_X*2) {
+                setSpritePosX(getSpritePosX() + SPRITE_SIZE_X); //moves to last walking position on sprite
+            } else {
+                setSpritePosX(SPRITE_SIZE_X);   //moves to first walking position on sprite
+            }
+        } 
+    }
+    else if (state[SDL_SCANCODE_DOWN] || state[SDL_SCANCODE_S]) {
+        if (getSpritePosX() == 0) {
+            setSpritePosX(SPRITE_SIZE_X);   //stops lag when taking first step
+        }
+        else if ((int)frameCount % FRAME_COUNT_WALK == 0) {
```

Comment:
This indentation actually managed to follow none of any other accepted styles
Take a look at the examples please